File: src/Qwack.Transport/TransportObjects/MarketData/Curves/TO_PriceCurve.cs
using ProtoBuf;

namespace Qwack.Transport.TransportObjects.MarketData.Curves
{
    [ProtoContract]
    public class TO_PriceCurve
    {
        [ProtoMember(1)]
        public TO_BasicPriceCurve BasicPriceCurve { get; set; }
        [ProtoMember(2)]
        public TO_ContangoPriceCurve ContangoPriceCurve { get; set; }
        [ProtoMember(3)]
        public TO_BasisPriceCurve BasisPriceCurve { get; set; }
        [ProtoMember(4)]
        public TO_ConstantPriceCurve ConstantPriceCurve { get; set; }
        [ProtoMember(5)]
        public TO_EquityPriceCurve EquityPriceCurve { get; set; }
    }
}

File: src/Qwack.Transport/GeneratedTypeScript/TO_PriceCurve.ts

import { TO_BasicPriceCurve } from './TO_BasicPriceCurve';
import { TO_ContangoPriceCurve } from './TO_ContangoPriceCurve';
import { TO_BasisPriceCurve } from './TO_BasisPriceCurve';
import { TO_ConstantPriceCurve } from './TO_ConstantPriceCurve';

export interface TO_PriceCurve  {
	 basicPriceCurve: TO_BasicPriceCurve;
	 contangoPriceCurve: TO_ContangoPriceCurve;
	 basisPriceCurve: TO_BasisPriceCurve;
	 constantPriceCurve: TO_ConstantPriceCurve;
}


